src/api_server.cc

```diff
@@ -16,106 +16,45 @@
 
 #include "api_server.h"
 
-#define BOOST_NETWORK_ENABLE_HTTPS
-#include <boost/network/protocol/http/client.hpp>
-#include <boost/network/protocol/http/server.hpp>
 #include <boost/range/irange.hpp>
-#include <memory>
-#include <ostream>
-#include <thread>
 
-#include "environment.h"
-#include "format.h"
+#include "configuration.h"
 #include "http_common.h"
-#include "json.h"
 #include "logging.h"
-#include "oauth2.h"
-#include "time.h"
-
-namespace http = boost::network::http;
+#include "store.h"
 
 namespace google {
 
-MetadataAgent::Metadata MetadataAgent::Metadata::IGNORED() {
-  return MetadataAgent::Metadata();
-}
-
-
-class MetadataApiServer {
- public:
-  MetadataApiServer(const MetadataAgent& agent, int server_threads,
-                    const std::string& host, int port);
-  ~MetadataApiServer();
-
- private:
-  class Handler;
-  using HttpServer = http::server<Handler>;
-  class Handler {
-   public:
-    Handler(const MetadataAgent& agent);
-    void operator()(const HttpServer::request& request,
-                    std::shared_ptr<HttpServer::connection> conn);
-    void log(const HttpServer::string_type& info);
-   private:
-    const MetadataAgent& agent_;
-  };
-
-  Handler handler_;
-  HttpServer server_;
-  std::vector<std::thread> server_pool_;
-};
-
-class MetadataReporter {
- public:
-  MetadataReporter(MetadataAgent* agent, double period_s);
-  ~MetadataReporter();
-
- private:
-  // Metadata reporter.
-  void ReportMetadata();
-
-  // Send the given set of metadata.
-  void SendMetadata(
-      std::map<MonitoredResource, MetadataAgent::Metadata>&& metadata)
-      throw (boost::system::system_error);
-
-  MetadataAgent* agent_;
-  Environment environment_;
-  OAuth2 auth_;
-  // The reporting period in seconds.
-  time::seconds period_;
-  std::thread reporter_thread_;
-};
-
-MetadataApiServer::Handler::Handler(const MetadataAgent& agent)
-    : agent_(agent) {}
+MetadataApiServer::Handler::Handler(const MetadataAgentConfiguration& config,
+                                    const MetadataStore& store)
+    : config_(config), store_(store) {}
 
 void MetadataApiServer::Handler::operator()(const HttpServer::request& request,
                                             std::shared_ptr<HttpServer::connection> conn) {
   static const std::string kPrefix = "/monitoredResource/";
   // The format for the local metadata API request is:
   //   {host}:{port}/monitoredResource/{id}
-  if (agent_.config_.VerboseLogging()) {
+  if (config_.VerboseLogging()) {
     LOG(INFO) << "Handler called: " << request.method
               << " " << request.destination
               << " headers: " << request.headers
               << " body: " << request.body;
   }
   if (request.method == "GET" && request.destination.find(kPrefix) == 0) {
```

Comment:
It'd be awesome to call something like `MonitoredResourceHandler` here, so that we can easily start adding other handlers based on method / path such as `HealthcheckHandler`, then, each handler would be easily testable as well.

Reply:
Good idea, bad timing. This is supposed to be a refactoring with a specific purpose — I'll do this in a follow-up PR if you don't mind.